service/resource/kmskeyv1/delete_test.go

```diff
@@ -1 +1,143 @@
 package kmskeyv1
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/clustertpr"
+	"github.com/giantswarm/clustertpr/spec"
+	"github.com/giantswarm/micrologger/microloggertest"
+)
+
+func Test_newDelete(t *testing.T) {
+	clusterTpo := awstpr.CustomObject{
+		Spec: awstpr.Spec{
+			Cluster: clustertpr.Spec{
+				Cluster: spec.Cluster{
+					ID: "test-cluster",
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		currentState  KMSKeyState
+		desiredState  KMSKeyState
+		expectedAlias string
+		description   string
+	}{
+		{
+			description:   "current and desired state empty, expected empty",
+			currentState:  KMSKeyState{},
+			desiredState:  KMSKeyState{},
+			expectedAlias: "",
+		},
+		{
+			description:  "current state empty, desired state not empty, expected empty",
+			currentState: KMSKeyState{},
+			desiredState: KMSKeyState{
+				KeyAlias: "desired",
+			},
+			expectedAlias: "",
+		},
+		{
+			description: "current state not empty, desired state not empty but different, expected current state",
+			currentState: KMSKeyState{
+				KeyAlias: "current",
+			},
+			desiredState: KMSKeyState{
+				KeyAlias: "desired",
+			},
+			expectedAlias: "current",
+		},
+		{
+			description: "current state not empty, desired state not empty but equal, expected desired state",
+			currentState: KMSKeyState{
+				KeyAlias: "current",
+			},
+			desiredState: KMSKeyState{
+				KeyAlias: "current",
+			},
+			expectedAlias: "current",
+		},
+	}
+
+	var err error
+	var newResource *Resource
+	resourceConfig := DefaultConfig()
+	resourceConfig.Clients = Clients{}
+	resourceConfig.Logger = microloggertest.New()
+	newResource, err = New(resourceConfig)
+	if err != nil {
+		t.Error("expected", nil, "got", err)
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
```

Comment:
I really like that we're using subtests here. We should use them more widely.

Reply:
Indeed, I try to use them as much as i can :)